plugins/modules/win_zip.ps1

```diff
@@ -0,0 +1,57 @@
+#!powershell
+
+# Copyright: (c) 2021, Kento Yagisawa <thel.vadam2485@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+$ErrorActionPreference = "Stop"
+
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+$src = Get-AnsibleParam -obj $params -name "src" -type "path" -failifempty $true
+$dest = Get-AnsibleParam -obj $params -name "dest" -type "path" -failifempty $true
+$force = Get-AnsibleParam -obj $params -name "overwrite" -type "bool" -default $false
+
+$result = @{
+    changed = $false
+    dest = $dest -replace '\$',''
+    src = $src -replace '\$',''
+}
```

Comment:
We should try and use the new module wrapper for this instead of the legacy one. There are numerous benefits you get from using that over the older style

```suggestion
#AnsibleRequires -CSharpUtil Ansible.Basic

$spec = @{
    options = @{
        src = @{ type = 'path'; required = $true }
        dest = @{ type = 'path'; required = $true }
        overwrite = @{ type = 'bool'; default = $false }
    }
    supports_check_mode = $true
}
$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)

$src = $module.Params.src
$dest = $module.Params.dest
$force = $module.Params.overwrite
```

Some of the features are

* Automatic module invocation details when run with `-vvv`
    * You no longer need to, or should, return `src` and `dest` in your return value
* Better option validation and error messages around that
* Event logging when running the modules

Some things you do need to change with the newer format

* `Fail-Json` becomes `$module.FailJson("My message here")`
* `Exit-Json` become `$module.ExitJson()`
* Getting check mode is now `$module.CheckMode
* Settings results is now done on `$module.Results.foo = 'bar'`, e.g. `$module.Results.changed = $true`

Reply:
Okay I will rewrite the module with the new module wrapper